tools/platform/performance-counter.h

```diff
@@ -0,0 +1,30 @@
+#ifndef PLATFORM_PERFORMANCE_COUNTER_H
+#define PLATFORM_PERFORMANCE_COUNTER_H
+
+#include <chrono>
+
+namespace platform
+{
+typedef std::chrono::high_resolution_clock::time_point TimePoint;
+typedef std::chrono::high_resolution_clock::duration Duration;
```

Comment:
It seems a little weird that there is code for a `std::function` replacement earlier in this PR, but then here we are just relying on `std::chrono` directly.

I have a general preference for *not* relying on services from `std` if we can get them from native platform APIs instead without too much trouble. If this is really the best way to get a high-performance timer that we can use everywhere, then so be it.